src/snowflake/connector/connection_diagnostic.py

```diff
@@ -0,0 +1,365 @@
+import ipaddress
+import json
+import os
+import re
+import requests
```

Comment:
You should be using the vendored `requests`. Extra points if you can just use the vendored `urllib3` instead